### PR TITLE
fixed lambda LR saving bug

### PR DESCRIPTION
**Pull request by humzaiqbal**
## Description of proposed changes
This PR fixes a probelm with the LambaLR scheduler in which label models trained with it can't be serialized properly because it uses a lambda function which pickle doesn't save. I fixed it by changing the scheduler to be class based using this as reference https://stackoverflow.com/questions/52758051/how-to-save-lambdalr-scheduler-in-pytorch-with-lambda-function
## Related issue(s)
https://github.com/snorkel-team/snorkel/issues/1658
Fixes # (issue)

## Test plan
I added a UT where I set a label model using this scheduler and ensured that now it can be saved
## Checklist

Need help on these? Just ask!

*  ✅ I have read the **CONTRIBUTING** document.
*  ✅ I have updated the documentation accordingly.
*  ✅  I have added tests to cover my changes.
*  ✅  I have run `tox -e complex` and/or `tox -e spark` if appropriate.
*  ✅  All new and existing tests passed.

### CODE REVIEW

File: snorkel/labeling/model/label_model.py [hunk 1]
```diff
@@ -675,7 +675,7 @@ def _set_lr_scheduler(self) -> None:
                 total_steps - self.warmup_steps
             )
             lr_scheduler = optim.lr_scheduler.LambdaLR(  # type: ignore
-                self.optimizer, linear_decay_func
+                self.optimizer, lr_lambda=LRPolicy(self.warmup_steps, total_steps)
```
Comment (bhancock8) [inline snorkel/labeling/model/label_model.py:678]: Is linear_decay_func still being used elsewhere or should it be dropped form that file altogether?

File: test/labeling/model/test_label_model.py [hunk 2]
```diff
@@ -548,6 +549,23 @@ def test_symmetry_breaking_multiclass(self):
...
+        label_model.fit(
+            L_train=L_train,
+            n_epochs=500,
+            lr=0.001,
+            lr_scheduler="linear",
+            optimizer="adam",
+            log_freq=100,
+            seed=123,
+        )
+        label_model.save("./test_save.pickle")
```
Comment (bhancock8) [inline test/labeling/model/test_label_model.py:566]: Not a bad idea to include a comment why this test exists (i.e., it's sort of strange to save a model then delete it without reloading or doing anything else with it). Can mention that this model was not previously pickleable due to a lambda function in the schedule (leave breadcrumbs for future readers of the code)
